### PR TITLE
[5.4] Add transpose to \Illuminate\Support\Collection

### PR DESCRIPTION
**Pull request by DrowningElysium**
I was hoping to get the transpose method in the Collection class as this would make formatting data from a form easier. 
An example in how this is easier can be seen in [this post](https://adamwathan.me/2016/04/06/cleaning-up-form-input-with-transpose/) of Adam Wathan.

I have taken some inspiration from #14433, so some credits go to the people who contributed to it.

### CODE REVIEW

File: src/Illuminate/Support/Collection.php [hunk 1]
```diff
@@ -1190,6 +1190,33 @@ public function transform(callable $callback)
     }
 
     /**
+     * Rotate the multidimensional array,
+     * turning the rows into columns and the columns into rows.
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Collection.php]: Please format this correctly.

File: src/Illuminate/Support/Collection.php [hunk 2]
```diff
@@ -1190,6 +1190,33 @@ public function transform(callable $callback)
...
+     *
+     * @return static
+     *
+     * @throws \LengthException
+     */
+    public function transpose()
+    {
+        if (count($this->items, COUNT_RECURSIVE) <= count($this->items) * 2) {
+            throw new \LengthException('Collection is not multidimensional with at least one child each!');
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Collection.php]: Please import this,

**Comment by GrahamCampbell** [comment]
The transpose function has already been rejected, because it can be added as a macro?

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by DrowningElysium** [comment]
I thought there were some issues with the implementation or something. That's why I made a PR as I think I fixed those.

**Comment by taylorotwell** [comment]
Going to leave this as a macro option for now.

**Pull request closed without merging**